tests/convert_utf8_to_latin1_with_errors_tests.cpp

```diff
@@ -0,0 +1,223 @@
+
+ TEST(too_large_input) {
+  uint32_t seed{1234};
+  simdutf::tests::helpers::RandomIntRanges random({{0xff, 0x10FFFF}}, seed);
+
+auto getUtf8SequenceLength = [](char byte) {
+    if ((byte & 0b11100000) == 0b11000000) { // 2 byte UTF-8 header
+        return 2;
+    }
+    else if ((byte & 0b11110000) == 0b11100000) { // 3 byte UTF-8 header
+        return 3;
+    }
+    else if ((byte & 0b11111000) == 0b11110000) { // 4 byte UTF-8 header
+        return 4;
+    }
+    else {
+        return 1; // 1 byte UTF-8 sequence (ASCII)
+    }
+};
+
+  for(size_t trial = 0; trial < trials; trial++) {
+    transcode_utf8_to_latin1_test_base test(random, fix_size);
+    for (int i = 0; i < fix_size; i++) {
+      auto byte_number = getUtf8SequenceLength(test.input_utf8[i]);
+      if(byte_number != 1) { 
+
+        auto procedure = [&implementation, &i](const char* utf8, size_t size, char* latin1) -> size_t {
+          simdutf::result res = implementation.convert_utf8_to_latin1_with_errors(utf8, size, latin1);
+          ASSERT_EQUAL(res.error,simdutf::error_code::TOO_LARGE);
+          ASSERT_EQUAL(res.count,i); 
+          return 0;
+        };
+
+        ASSERT_TRUE(test(procedure)); // no conversion should take place
+
+        // does the same as the conditional above: e.g. replace a 4 byte by a '*' ASCII character once its done.
+        for(auto it = test.input_utf8.begin(); it != test.input_utf8.end(); ++it) {
+            if(std::distance(it, test.input_utf8.end()) >= byte_number) { 
+                std::fill_n(it, byte_number, 0x2a);
+            }
+        }
+
+        }
+    }
+  }
+} 
+
+TEST(header_bits_error) {
+  uint32_t seed{1234};
+  simdutf::tests::helpers::RandomIntRanges random({{0x0000, 0xff}}, seed);
+
+  for(size_t trial = 0; trial < trials; trial++) {
+    transcode_utf8_to_latin1_test_base test(random, fix_size);
+
+    for (int i = 0; i < fix_size; i++) {
+
+      if((test.input_utf8[i] & 0b11000000) != 0b10000000) {  // Only process leading bytes
+        auto procedure = [&implementation, &i](const char* utf8, size_t size, char* latin1) -> size_t {
+          simdutf::result res = implementation.convert_utf8_to_latin1_with_errors(utf8, size,latin1);
+          ASSERT_EQUAL(res.error, simdutf::error_code::HEADER_BITS);
+          ASSERT_EQUAL(res.count, i);
+          return 0;
+        };
+        const unsigned char old = test.input_utf8[i];
+        test.input_utf8[i] = uint8_t(0b11111000);
+        ASSERT_TRUE(test(procedure));
+        test.input_utf8[i] = old;
+      }
+    }
+  }
+}
+
+TEST(too_short_error) {
+  uint32_t seed{1234};
+  simdutf::tests::helpers::RandomIntRanges random({{0x00, 0xff}}, seed);
+  for(size_t trial = 0; trial < trials; trial++) {
+    transcode_utf8_to_latin1_test_base test(random, fix_size);
+    int leading_byte_pos = 0;
+    for (int i = 0; i < fix_size; i++) {
+
+      if((test.input_utf8[i] & 0b1100000) == 0b10000000) {  // Only process continuation bytes by making them leading bytes
```

Comment:
Something is wrong here: `test.input_utf8[i] & 0b1100000) == 0b10000000`. I think that you might be missing a trailing zero.